pennylane/beta/tapes/cv_param_shift.py

```diff
@@ -0,0 +1,458 @@
+        tape consisting of *only* expectation values of observables.
+
+        .. note::
+
+            The 2nd order method can handle also first order observables, but
+            1st order method may be more efficient unless it's really easy to
+            experimentally measure arbitrary 2nd order observables.
+
+        Args:
+            idx (int): trainable parameter index to differentiate with respect to
+            device (.Device, .QubitDevice): a PennyLane device
+                that can execute quantum operations and return measurement statistics
+            params (list[Any]): the quantum tape operation parameters
+
+        Keyword Args:
+            force_order2 (bool): iff True, use the order-2 method even if not necessary
+
+        Returns:
+            array[float]: 1-dimensional array of length determined by the tape output
+            measurement statistics
+        """
+        grad_method = self._par_info[idx]["grad_method"]
+
+        if options.get("force_order2", False) or grad_method == "A2":
+            return self.parameter_shift_second_order(idx, device, params, **options)
+
+        return self.parameter_shift_first_order(idx, device, params, **options)
+
+    def parameter_shift_var(self, idx, device, params, **options):
+        r"""Partial derivative using the first-order or second-order parameter-shift rule of a tape
+        consisting of a mixture of expectation values and variances of observables.
+
+        Expectation values may be of first- or second-order observables,
+        but variances can only be taken of first-order variables.
+
+        .. warning::
+
+            This method can only be executed on devices that support the
+            :class:`~.PolyXP` observable.
+
+        Args:
+            idx (int): trainable parameter index to differentiate with respect to
+            device (.Device, .QubitDevice): a PennyLane device
+                that can execute quantum operations and return measurement statistics
+            params (list[Any]): the quantum tape operation parameters
+
+        Keyword Args:
+            force_order2 (bool): iff True, use the order-2 method even if not necessary
+
+        Returns:
+            array[float]: 1-dimensional array of length determined by the tape output
+            measurement statistics
+        """
+        # pylint: disable=protected-access
+
+        if "PolyXP" not in device.observables:
+            # If the device does not support PolyXP, must fallback
+            # to numeric differentiation.
+            warnings.warn(
+                f"The device {device.short_name} does not support "
+                "the PolyXP observable. The analytic parameter-shift cannot be used for "
+                "second-order observables; falling back to finite-differences.",
+                UserWarning,
+            )
+            return self.numeric_pd(idx, device, params, **options)
+
+        temp_tape = self.copy()

```

Comment:
Since the second-order method is so dependent on the circuit graph, rather than attempting to modify `self` directly, it is cleaner to just copy the tape. This is because mutating the tapes existing measurements will not cause the equivalent mutation in the graph, since the graph has already been created!